From 024053671f9f02caa544eda6e076739cf04c451b Mon Sep 17 00:00:00 2001
From: Wenchao Hao <haowenchao2@huawei.com>
Date: Fri, 7 Apr 2023 15:04:05 +0800
Subject: [PATCH 0009/1646] [Huawei] kabi: Define KABI_RESERVE_VENDOR for
 VENDOR to reserve kabi field

Offering: HULK
euleros inclusion
category: bugfix
bugzilla: 188541

--------------------------------

Define a new macro KABI_RESERVE_VENDOR to tell this kabi field is
used by downstream VENDOR like EulerOS. The KABI_RESERVE_VENDOR
has the same function as KABI_RESERVE.

Signed-off-by: Wenchao Hao <haowenchao2@huawei.com>
---
 include/linux/kabi.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/include/linux/kabi.h b/include/linux/kabi.h
index 2a1cf02fbb15..f33171e00ee7 100644
--- a/include/linux/kabi.h
+++ b/include/linux/kabi.h
@@ -331,6 +331,9 @@
  *   Adds a new field to an enumeration type.  This must always be added to
  *   the end of the enum.  Before using this macro, make sure this is actually
  *   safe to do.
+ * KABI_RESERVE_VENDOR
+ *   This macro has the same function as KABI_RESERVE, while declaring that
+ *   this field is used by specific VENDOR, such as EulerOS.
  */
 #ifdef __GENKSYMS__
 
@@ -441,6 +444,8 @@
  * Simple wrappers to replace standard openEuler reserved elements.
  */
 # define KABI_USE(n, _new)		KABI_REPLACE(_KABI_RESERVE(n), _new)
+
+# define KABI_RESERVE_VENDOR(n)		KABI_RESERVE(n)
 /*
  * Macros for breaking up a reserved element into two smaller chunks using
  * an anonymous struct inside an anonymous union.
-- 
2.17.1

